source/IOSResolver/src/IOSResolver.cs

```diff
@@ -1799,26 +1799,44 @@ public enum CocoapodsIntegrationMethod {
         string line;
 
         // We are only interested in capturing the dependencies "Pod depName, depVersion", inside
-        // of the specific target. However there can be nested targets such as for testing, so we're
+        // of the targets in XcodeTargetNames. However there can be nested targets such as for testing, so we're
         // counting the depth to determine when to capture the pods. Also we only ever enter the
         // first depth if we're in the exact right target.
         int capturingPodsDepth = 0;
+        string matchedTarget = null;
         var sources = new List<string>();
+        var targetsToCheck = new Dictionary<string, string>();;
```

Comment:
Ah, as mentioned on the prior match set I think a RegEx would simplify this quite a bit.